include/triton/Dialect/TritonGPU/IR/TritonGPUOps.td

```diff
@@ -276,4 +276,26 @@ def TTG_UpcastMXFPOp : TTG_Op<"upcast_mxfp", [Pure, DeclareOpInterfaceMethods<In
   }];
 }
 
+// Allocate global memory
+def TTG_GlobalScratchAllocOp : TTG_Op<"global_scratch_alloc", [MemoryEffects<[MemAlloc<GlobalMemory>]>]> {
+  let summary = "allocate a global memory buffer";
+  let description = [{
+    This operation allocates a buffer in global memory.
```

Comment:
extra nit: you may want to mention that the buffer is private to a given program instance